sources/osgStats/Stats.js

```diff
@@ -206,28 +274,125 @@ utils.createPrototypeObject(Stats, {
+                var geometry = this._bufferStats.getGeometry();
+                geometry.setName('StatsGeometry');
+                geometry.setNodeMask(~PICKING_NODEMASK);
+                node.addChild(geometry);
                 texture.setImage(canvas);
+                // invalidate the bounding box, we dont want it
+                // to be used
+                geometry.setBound(new BoundingBox());
             }.bind(this)
         );
 
         node.getOrCreateStateSet().setAttributeAndModes(createShader());
         node.getOrCreateStateSet().setAttributeAndModes(new CullFace(CullFace.DISABLE));
-        node
-            .getOrCreateStateSet()
-            .setAttributeAndModes(new BlendFunc('SRC_ALPHA', 'ONE_MINUS_SRC_ALPHA'));
+        var blendFunc = new BlendFunc('SRC_ALPHA', 'ONE_MINUS_SRC_ALPHA');
+        node.getOrCreateStateSet().setAttributeAndModes(blendFunc);
         node.getOrCreateStateSet().setAttributeAndModes(new Depth(Depth.DISABLE));
         node.getOrCreateStateSet().setTextureAttributeAndModes(0, texture);
+
+        var canvas = this._viewer.getGraphicContext().canvas;
+
+        this._canvas = canvas;
+
+        canvas.addEventListener('mousemove', this.onMouseMove.bind(this));
+        canvas.addEventListener('mousedown', this.onMouseDown.bind(this));
+        canvas.addEventListener('mouseup', this.onMouseUp.bind(this));
+        canvas.addEventListener('mouseout', this.onMouseUp.bind(this));
+
+        canvas.addEventListener('touchmove', this.onMouseMove.bind(this));
+        canvas.addEventListener('touchstart', this.onMouseDown.bind(this));
+        canvas.addEventListener('touchend', this.onMouseUp.bind(this));
+        canvas.addEventListener('touchcancel', this.onMouseUp.bind(this));
+    },
+    onMouseDown: function(e) {
+        var hits = this.computeNearestIntersection(e);
+        this._onStats = !!hits.length;
+
+        e.preventDefault();
+        this._eventMouse.setEnable(false);
+        this._eventTouch.setEnable(false);
+        mat4.copy(this._startTransformation, this._nodeTransform.getMatrix());
+        getCanvasCoord(this._dragStart, e);
+        this._dragStop[1] = this._dragStart[1];
     },
+    onMouseMove: (function() {
+        return function(e) {
+            if (!this._onStats) return;
+
+            getCanvasCoord(this._dragStop, e);
+            var dy = this._dragStop[1] - this._dragStart[1];
+            mat4.translate(
+                this._nodeTransform.getMatrix(),
+                this._startTransformation,
+                vec3.fromValues(0, dy, 0)
+            );
+        };
+    })(),
+    onMouseUp: function(e) {
+        this._onStats = false;
+        this._eventMouse.setEnable(true);
+        this._eventTouch.setEnable(true);
+        this._eventMouse.mouseup(e);
+    },
+
+    computeNearestIntersection: (function() {
+        var coord = vec2.create();
+        var lsi = new LineSegmentIntersector();
```

Comment:
For stats picking should be good to stop at first intersection if we disable intersection in the rest of the graph. With this PR I think scene is being intersected, am I right?